pkg/bpf/bpf_linux.go

```diff
@@ -662,6 +663,8 @@ func TestDummyProg(progType ProgType, attachType uint32) error {
 			return errno
 		}
 		return nil
+	} else if errors.Is(errno, unix.EPERM) {
+		log.WithError(errno).Warningf("Cilium cannot load bpf programs. Security profiles like SELinux may be restricting permissions.")
```

Comment:
This error message will get printed in addition to the other log message. It would be more consistent to add this condition at the same place where the "BPF host reachable services for UDP needs kernel..." message is printed, maybe by doing the check via errors.Is(err, unix.EPERM) at that location rather than doing the direct errno check at this point.

Could you take a look at how this could be better integrated with the log error handling in the parent function `initKubeProxyReplacementOptions()` or above instead, where this function is called from?

Reply:
how about this?